src/coreclr/vm/class.cpp

```diff
@@ -124,25 +124,30 @@ void EEClass::Destruct(MethodTable * pOwningMT)
     if (IsDelegate())
     {
         DelegateEEClass* pDelegateEEClass = (DelegateEEClass*)this;
-
-        if (pDelegateEEClass->m_pStaticCallStub)
+        for (Stub* pThunk : {pDelegateEEClass->m_pStaticCallStub, pDelegateEEClass->m_pInstRetBuffCallStub})
         {
-            // Collect data to remove stub entry from StubManager if
-            // stub is deleted.
-            BYTE* entry = (BYTE*)pDelegateEEClass->m_pStaticCallStub->GetEntryPoint();
-            UINT length = pDelegateEEClass->m_pStaticCallStub->GetNumCodeBytes();
-
-            ExecutableWriterHolder<Stub> stubWriterHolder(pDelegateEEClass->m_pStaticCallStub, sizeof(Stub));
-            BOOL fStubDeleted = stubWriterHolder.GetRW()->DecRef();
-            if (fStubDeleted)
+            if (pThunk == nullptr)
+                continue;
+
+            _ASSERTE(pThunk->IsShuffleThunk());
+
+            if (pThunk->HasExternalEntryPoint()) // IL thunk
             {
-                StubLinkStubManager::g_pManager->RemoveStubRange(entry, length);
+                pThunk->DecRef();
+            }
+            else
+            {
+                // Collect data to remove stub entry from StubManager if stub is deleted.
+                BYTE* entry = (BYTE*)pThunk->GetEntryPoint();
+                UINT length = pThunk->GetNumCodeBytes();
+
+                ExecutableWriterHolder<Stub> stubWriterHolder(pThunk, sizeof(Stub));
+                BOOL fStubDeleted = stubWriterHolder.GetRW()->DecRef();
+                if (fStubDeleted)
+                {
+                    StubLinkStubManager::g_pManager->RemoveStubRange(entry, length);
+                }
```

Comment:
```suggestion
                ExecutableWriterHolder<Stub> stubWriterHolder(pThunk, sizeof(Stub));
                stubWriterHolder.GetRW()->DecRef();
```
I do not think the `RemoveStubRange` does anything useful here. The memory occupied by stub is not going to be released anyway. It is probably a left-over from a different allocation scheme.

You can also delete the `RemoveStubRange` method since this is the only place where it is used.

Reply:
done

Reply:
Looks like this was the only place where start-end arguments from RangeList::RemoveRanges[Worker] were used. Also deleted.